<?xml version="1.0" encoding="UTF-8"?>
<mule xmlns="http://www.mulesoft.org/schema/mule/core"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns:spring="http://www.springframework.org/schema/beans" 
    xmlns:vm="http://www.mulesoft.org/schema/mule/vm"
    xsi:schemaLocation="
       http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-current.xsd
       http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
       http://www.mulesoft.org/schema/mule/vm http://www.mulesoft.org/schema/mule/vm/current/mule-vm.xsd">

    <vm:connector name="vmConnector"/>

	
    <flow name="HelloComponent">            
		<inbound-endpoint address="vm://hello.in" exchange-pattern="request-response"/>            
        <component class="org.mule.test.integration.routing.nested.BindingExceptionOnInterfaceMethodTestCase$ExceptionThrowingService" />
            
		<!-- No need to log the exceptions happening in the component - we know there will be one -->
        <custom-exception-strategy class="org.mule.functional.functional.QuietExceptionStrategy"/>
	</flow>

	<flow name="InvokerComponent">            
		<vm:inbound-endpoint path="invoker.in" exchange-pattern="request-response"/>            
		<component>
			<singleton-object class="org.mule.test.integration.routing.nested.BindingExceptionOnInterfaceMethodTestCase$Component" />
			<binding interface="org.mule.test.integration.routing.nested.BindingExceptionOnInterfaceMethodTestCase$BindigInterface">
				<outbound-endpoint address="vm://hello.in" exchange-pattern="request-response" />
			</binding>
		</component>                                    
	</flow>
    
</mule>
